spring-cloud-task-core/src/main/java/org/springframework/cloud/task/repository/dao/MapTaskExecutionDao.java

```diff
@@ -101,6 +109,19 @@ public long getTaskExecutionCount(String taskName) {
 		return new ArrayList<String>(result);
 	}
 
+	@Override
+	public Page<TaskExecution> findAll(Pageable pageable) {
+		Set<TaskExecution> sortedSet = getTaskExecutionTreeSet();
+		sortedSet.addAll(taskExecutions.values());
+		List<TaskExecution> result = new ArrayList<>(sortedSet);
```

Comment:
Since order matters to us, why not use `ConcurrentSkipListMap` instead of creating a `TreeSet` every time we ask for the results?  Then you (I believe) should be able to go straight from the `Map` to the `List` without this `Set` step...